lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -244,8 +226,119 @@ def _run(self, params, concierge_params=None):
+    #         # Get Method Version via Git Commit
+    #         if method not in mv_cache and service_ver not in mv_cache[method]:
+    #             mv_cache[method][service_ver]= self._get_module_git_commit(method=method,
+    #                                                                        service_ver=service_ver)
+    #
+    #     return CatalogUtils
+
+    # TODO delete this after checking it in
+    # def _get_cached_catalog_resources(
+    #     self, job_param_set: List[dict]
+    # ) -> JobResourceMappings:
+    #     # Check workspace objects for each of the jobs
+    #     workspace_check_time = time.time()
+    #     [
+    #         self._check_ws_objects(param.get("source_ws_objects"))
+    #         for param in job_param_set
+    #     ]
+    #     self.logger.debug(
+    #         f"Took {time.time() - workspace_check_time} to check all workspace objects"
+    #     )
+    #     #Get a mapping of condor resources
+    #     catalog_check_time = time.time()
+    #     condor_resources_mapping = self.catalog_utils.get_mass_resources(
+    #         job_param_set=job_param_set, condor=self.condor
+    #     )
+    #     self.logger.debug(
+    #         f"Took {time.time() - catalog_check_time} to get all condor resources"
+    #     )
+    #     # Get a mapping of method git versions. If one is not valid, throw the catalog error #TODO Test this
+    #     catalog_check_time_git = time.time()
+    #
+    #
+    #     return JobResourceMappings(
+    #         condor_resources=condor_resources_mapping,
+    #         method_service_versions=service_versions_mapping,
+    #     )
+
+    def _evaluate_job_params_set(self, job_param_set, parent_job):
+        # Fail early before job submission if possible
+        # Make sure clientgroups and condor resources are set correctly for each job,
+        # or raise an exception if something goes wrong
+        for job_params in job_param_set:
+            self.catalog_utils.get_git_commit_version(job_params=job_params)
+            self.catalog_utils.get_condor_resources(
+                job_params=job_params, condor=self.condor
+            )
+            # Don't allow differing parent jobs, and ensure job always runs with the parent
+            if (
+                "parent_job_id" in job_params
+                and job_params["parent_job_id"] != parent_job.id
+            ):
+                raise MultipleParentJobsException(
+                    "Launching child jobs with differing parents is not allowed"
+                )
+
+    def _run_batch(self, parent_job: Job, job_param_set: List[Dict]) -> List:
+        # Prepare jobs, fail early if possible
+        self._evaluate_job_params_set(job_param_set, parent_job)
+
+        child_job_ids = []
+        # Initiatialize Job Records
+        for job_params in job_param_set:
+            job_params["parent_job_id"]: parent_job.id
+            resources = self.catalog_utils.get_condor_resources(
```

Comment:
Can the condor resources be different per child job? I'm not even sure if child jobs running different apps are allowed

Reply:
If they're the same, could you return the resources from _evaluate... and reuse them?

Reply:
Child jobs running different apps are allowed for import and batch.
The resources are being returned from the resources cache and being re-used, eliminating the need to look them up from the catalog for each job, e.g., running the same app in the same batch gets looked up once and now info is taken from the cache. this has the added benefit of each child job using the same resources and git commit if the catalog is updated during a job run.

Reply:
gotcha